src/targets/gpu/gemm_impl.cpp

```diff
@@ -97,64 +104,124 @@ void gemm_impl(context& ctx,
+                               0,
+                               flag);
         }
         else
         {
-            rocblas_invoke(&rocblas_gemm_strided_batched_ex,
-                           ctx.get_stream().get_rocblas(),
-                           transb ? rocblas_operation_transpose : rocblas_operation_none,
-                           transa ? rocblas_operation_transpose : rocblas_operation_none,
-                           n,
-                           m,
-                           k,
-                           &alpha_r,
-                           to_pointer(args.at(1)),
-                           arg_type,
-                           ldb,
-                           k * n,
-                           to_pointer(args.at(0)),
-                           arg_type,
-                           lda,
-                           m * k,
-                           &beta_r,
-                           to_pointer(args[2]),
-                           output_type,
-                           ldc,
-                           m * n,
-                           is_3inputs ? to_pointer(args[3]) : to_pointer(args[2]),
-                           output_type,
-                           ldc,
-                           m * n,
-                           num_matrices,
-                           compute_type,
-                           rocblas_gemm_algo_standard,
-                           0,
-                           flag);
+            if(compute_fp32)
+                rocblas_invoke(&rocblas_gemm_strided_batched_ex,
+                               ctx.get_stream().get_rocblas(),
+                               transb ? rocblas_operation_transpose : rocblas_operation_none,
+                               transa ? rocblas_operation_transpose : rocblas_operation_none,
+                               n,
+                               m,
+                               k,
+                               &alpha,
+                               to_pointer(args.at(1)),
+                               arg_type,
+                               ldb,
+                               k * n,
+                               to_pointer(args.at(0)),
+                               arg_type,
+                               lda,
+                               m * k,
+                               &beta,
+                               to_pointer(args[2]),
+                               output_type,
+                               ldc,
+                               m * n,
+                               is_3inputs ? to_pointer(args[3]) : to_pointer(args[2]),
+                               output_type,
+                               ldc,
+                               m * n,
+                               num_matrices,
+                               compute_type,
+                               rocblas_gemm_algo_standard,
+                               0,
+                               flag);
```

Comment:
Can we combine this function call instead doing two calls? Maybe we can create the alpha/beta pointers before calling the function, etc.

Reply:
fixed